src/handshakes/abstract-handshake.ts

```diff
@@ -119,7 +118,9 @@ export abstract class AbstractHandshake {
   }
 
   protected getHash(a: bytes, b: bytes): bytes32 {
-    return SHA256.digest(Buffer.from([...a, ...b]));
+    return Buffer.from(
+      hash.sha256().update(Buffer.from([...a, ...b])).digest('hex')
```

Comment:
This hash is invalid. Buffer.from uses utf-8 encoding by default.

To reduce memory allocation:
```javascript
const hash = hash.sha256().update(Buffer.from([...a, ...b])).digest();
Buffer.from(hash.buffer, hash.offset, hash.length);
```
https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_arraybuffer_byteoffset_length